src/test/scala/com/wavesplatform/database/LevelDBWriterSpec.scala

```diff
@@ -154,6 +155,54 @@ class LevelDBWriterSpec extends FreeSpec with Matchers with WithDB with RequestG
       }
     }
 
+    "return Left if fromId argument is a non-existent transaction" in {
+      baseTest(time => preconditions(time.correctedTime())) { (writer, account) =>
+        val nonExistentTxId = GenesisTransaction.create(account, ENOUGH_AMT, 1).explicitGet().id()
+
+        val txs = writer
+          .addressTransactions(account.toAddress, Set(TransferTransactionV1.typeId), 3, Some(nonExistentTxId))
+
+        txs shouldBe Left(s"Transaction $nonExistentTxId does not exist")
+      }
+    }
+
+    "return txs in correct ordering starting from a given id" in {
+      baseTest(time => preconditions(time.correctedTime())) { (writer, account) =>
+        // using pagination
+        val firstTx = writer
+          .addressTransactions(account.toAddress, Set(TransferTransactionV1.typeId), 1, None)
+          .getOrElse(Seq.empty)
+          .head
+
+        val secondTx = writer
+          .addressTransactions(account.toAddress, Set(TransferTransactionV1.typeId), 1, Some(firstTx._2.id()))
+          .getOrElse(Seq.empty)
+          .head
+
+        // without pagination
+        val txs = writer
+          .addressTransactions(account.toAddress, Set(TransferTransactionV1.typeId), 2, None)
+          .getOrElse(Seq.empty)
+
+        txs shouldBe Seq(firstTx, secondTx)
+      }
+    }
+
+    "return an empty Seq when paginating from the last transaction" in {
+      baseTest(time => preconditions(time.correctedTime())) { (writer, account) =>
+        val txs = writer
+          .addressTransactions(account.toAddress, Set(TransferTransactionV1.typeId), 2, None)
+          .getOrElse(Seq.empty)
+
+        val txsFromLast = writer
+          .addressTransactions(account.toAddress, Set(TransferTransactionV1.typeId), 2, Some(txs.last._2.id()))
+          .getOrElse(txs) // should never return this value
```

Comment:
if this should not happen but fail instead, `.explicitGet()` may be more suitable

Reply:
+